src/currency-field/CurrencyField.ts

```diff
@@ -0,0 +1,389 @@
+        } catch (e) {
+            console.error('invalid locale or currency provided, fallback will be used', e);
+            currencyPartsMap = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD', currencyDisplay: 'narrowSymbol' })
+                .formatToParts(1000.0)
+                .map((c) => c.value);
+        }
+        // Get the currency symbol assuming it is the value at the first index of the array.
+        const currencySymbol = currencyPartsMap[0];
+
+        // Check to see if value at first index of the array is a number Some locales will have the currency symbol at the last index.
+        if (parseInt(currencySymbol)) {
+            this._currencyFormatSeparator = currencyPartsMap[CURRENCY_SEPARATOR.AMOUNT];
+            this._currencyCentsSeparator = currencyPartsMap[CURRENCY_SEPARATOR.CENTS];
+            this._currencySymbol = currencyPartsMap[currencyPartsMap.length - 1];
+        } else {
+            this._currencyFormatSeparator = currencyPartsMap[CURRENCY_SEPARATOR.INVERSE_AMOUNT];
+            this._currencyCentsSeparator = currencyPartsMap[CURRENCY_SEPARATOR.INVERSE_CENTS];
+            this._currencySymbol = currencySymbol;
+        }
+    }
+
+    _parseAmount(value: string): number {
+        let cleanValue = '';
+
+        for (let i = 0; i < value.length; i++) {
+            const character = value.charAt(i);
+            if (/\d/.test(character)) {
+                cleanValue += character;
+            }
+        }
+
+        if (cleanValue) {
+            return parseInt(cleanValue);
+        } else {
+            return null;
+        }
+    }
+
+    _parseCents(value: string): string {
+        let cleanValue = '';
+        for (let i = 0; i < value.length; i++) {
+            const character = value.charAt(i);
+            if (/\d/.test(character)) {
+                cleanValue += character;
+            }
+        }
+        return cleanValue;
+    }
+
+    _formatToCurrency(preFormattedValue: number): string {
+        if (!preFormattedValue) {
+            return '';
+        }
+        return this._currencyFormat.format(preFormattedValue);
+    }
+
+    // Format the internal value to a float.
+    _formatToFloat(formattedValue: string): string | number {
+        if (formattedValue.length > 0) {
+            // Some currency formats cents separator is not a '.'
+            if (this._currencyCentsSeparator !== '.' && formattedValue.includes(this._currencyCentsSeparator)) {
+                const lastIndex = formattedValue.lastIndexOf(this._currencyCentsSeparator);
+                formattedValue =
+                    formattedValue.substring(0, lastIndex).replaceAll(this._currencyFormatSeparator, '') +
+                    '.' +
+                    formattedValue.substring(lastIndex + 1);
+                return parseFloat(formattedValue);
+            } else {
+                const preFloatReplaceAll = formattedValue.replaceAll(this._currencyFormatSeparator, '');
```

Comment:
```suggestion
                const preFloatReplaceAll = formattedValue.replaceAll(this._currencyFormatSeparator, '').replaceAll(this._currencyCentsSeparator, '.');
```

This would have the same outcome as the if statement and would leave cleaner code.

Reply:
Implemented the suggested solution.